js/src/rive_advanced.mjs.d.ts

```diff
@@ -15,50 +20,121 @@ interface RiveOptions {
     AABB: AABB;
     SMIInput: typeof SMIInput;
     renderFactory: CanvasRenderFactory;
-  
+
     BlendMode: typeof BlendMode;
     FillRule: typeof FillRule;
     Fit: typeof Fit;
     RenderPaintStyle: typeof RenderPaintStyle;
     StrokeCap: typeof StrokeCap;
     StrokeJoin: typeof StrokeJoin;
-  
+
+    /**
+     * Loads a Rive file for the runtime and returns a Rive-specific File class
+     *
+     * @param buffer - Array buffer of a Rive file
+     * @returns A Promise for a Rive File class
+     */
     load(buffer: Uint8Array): Promise<File>;
-    makeRenderer(canvas: HTMLCanvasElement | OffscreenCanvas, useOffscreenRenderer?: boolean) : CanvasRenderer;
-    computeAlignment(fit: Fit, alignment: Alignment, frame: AABB, content: AABB): Mat2D;
+
+    /**
+     * Creates the renderer to draw the Rive on the provided canvas element
+     *
+     * @param canvas - Canvas to draw the Rive on
+     * @param useOffscreenRenderer - Option for those using the WebGL-variant of the Rive JS library. This uses an offscreen renderer to draw on the canvas, allowing for multiple Rives/canvases on a given screen. It is highly recommended to set this to `true` when using with the `@rive-app/webgl` package
+     * @returns A Rive CanvasRenderer class
+     */
+    makeRenderer(
+      canvas: HTMLCanvasElement | OffscreenCanvas,
+      useOffscreenRenderer?: boolean
+    ): CanvasRenderer;
+
+    /**
+     * Computes how the Rive is laid out onto the canvas
+     * @param {Fit} fit - Fit enum (i.e Fit.contain)
+     * @param alignment - Alignment enum (i.e Alignment.center)
+     * @param frame - AABB object representing the bounds of the canvas frame
+     * @param content - AABB object representing the bounds of what to draw the Rive onto (i.e an artboard's size)
+     * @returns Mat2D - A Mat2D view matrix
+     */
+    computeAlignment(
+      fit: Fit,
+      alignment: Alignment,
+      frame: AABB,
+      content: AABB
+    ): Mat2D;
     mapXY(matrix: Mat2D, canvasPoints: Vec2D): Vec2D;
+    /**
+     * A Rive-specific requestAnimationFrame function; this should be used over the global requestAnimationFrame function.
+     * @param cb - Callback function to call with an elapsed timestamp
+     * @returns number - An ID of the requestAnimationFrame request
+     */
     requestAnimationFrame(cb: (timestamp: DOMHighResTimeStamp) => void): number;
+    /**
+     * A Rive-specific cancelAnimationFrame function; this should be used over the global cancelAnimationFrame function.
+     * @param requestID - ID of the requestAnimationFrame request to cancel
+     */
     cancelAnimationFrame(requestID: number): void;
+    /**
+     * Debugging tool to showcase the FPS in the corner of the screen
+     */
```

Comment:
If you don't provide a callback does this method still add a DIV for you to display it in the corner?

Reply:
oh right! totally forgot thats a thing. this type needs to be altered to show the callback - but yeah, it still does that

Reply:
"passes the fps count to the callback instead"?

Or does it still make the new div if you give a callback?

Reply:
Nope, it doesn't! I'll clarify that